aptos-move/aptos-vm/src/block_executor/mod.rs

```diff
@@ -188,38 +205,46 @@ impl BlockExecutorTransactionOutput for AptosTransactionOutput {
             .clone()
     }
 
-    fn reads_needing_delayed_field_exchange(
-        &self,
-    ) -> BTreeMap<
-        <Self::Txn as BlockExecutableTransaction>::Key,
-        (
-            <Self::Txn as BlockExecutableTransaction>::Value,
-            Arc<MoveTypeLayout>,
-        ),
-    > {
+    fn reads_needing_delayed_field_exchange(&self) -> Vec<(StateKey, Arc<MoveTypeLayout>)> {
         self.vm_output
             .lock()
             .as_ref()
             .expect("Output to be set to get reads")
             .change_set()
-            .reads_needing_delayed_field_exchange()
-            .clone()
+            .resource_write_set()
+            .iter()
+            .flat_map(|(key, write)| {
+                if let AbstractResourceWriteOp::InPlaceDelayedFieldChange(change) = write {
+                    Some((key.clone(), change.layout.clone()))
+                } else {
+                    None
+                }
+            })
+            .collect()
     }
 
     fn group_reads_needing_delayed_field_exchange(
         &self,
-    ) -> BTreeMap<
+    ) -> Vec<(
         <Self::Txn as BlockExecutableTransaction>::Key,
```

Comment:
not sure but does it work to just have StateKey and WriteOp here? (and a few other places, just noticed that it seems not uniform)

Reply:
yeah, this is just what IDE automatically fills